### ISSUE TITLE
Всё сломалось

### ISSUE THREAD

**Issue by pozitronik** [pozitronik/CloudMailRu#314]
Похоже, поменялись какие-то внутренние урлы и форматы. Нужно разбираться.

**Comment by 1ehacom** [comment]
Теперь чтобы зайти нужно менять пароль часто, за этот год 3 раза уже сменил

**Comment by pozitronik** [comment]
> Теперь чтобы зайти нужно менять пароль часто, за этот год 3 раза уже сменил

Такого не наблюдаю. Да и помочь тут вряд ли смогу — это работает на стороне mail.ru.

**Comment by OmlineEditor** [comment]
Вы гворите за вот это при входе?
Login to userName@mail.ru
error: parsing auth token for userName@mail.ru

Я тоже не могу войти в аккаунт. Что делать?

**Comment by 1ehacom** [comment]
> Вы гворите за вот это при входе? Login to [userName@mail.ru](mailto:userName@mail.ru) error: parsing auth token for [userName@mail.ru](mailto:userName@mail.ru)
> 
> Я тоже не могу войти в аккаунт. Что делать?
Что делать? Войти через браузер и поменять пароль. Да и вообще 25 и 100 Гб уже превращают (отбирают) в 8 Гб. Так что и 1 тб могут забрать...

**Comment by midi-creator** [comment]
Зашёл через браузер, поменял пароль, не помогает((( error: parsing auth token

**Comment by midi-creator** [comment]
раньше ещё помогало переподключение у провайдера с выдачей нового серого айпи, сейчас не помогает(((( чтото наколдовали(((

**Comment by midi-creator** [comment]
спаситепааамогиттееее

**Comment by pozitronik** [comment]
> спаситепааамогиттееее

Да, у меня тоже сломалось. Обещать ничего не стану, но попробую разобраться.

**Comment by barb1973** [comment]
Через WebDAV работает.

**Comment by ViRT1ST** [comment]
Альтернативный вариант:

1. Скачайте плагин WebDAV с сайта Total Commander - https://www.ghisler.com/plugins.htm.
2. Закиньте его в папку "\Plugins\wfx\webdav" и включите его в Configuration -> Options -> Operation -> Plugins -> WFX.
3. Используйте инструкцию с mail.ru для подключения к WebDAV - https://help.mail.ru/cloud_web/app/webdav/
Нужно будет создать специальный пароль для подключения к Облаку  - https://account.mail.ru/user/2-step-auth/passwords/.
4. Заходите в плагин WebDAV (там же где находятся все сетевые плагины).
Жмете F7, придумываете имя для названия подключения, после чего появляется форма для подключения.
Вводите свои данные:

`Connect to: https://webdav.cloud.mail.ru`
`User name: ваше мыло`
`Password: ваш спец. пароль для облака`

Не забудьте поставить галочку "Secure Server (SSL)"

Авторизацию проходит через раз - то норм, то ошибка 401/403, но в целом работает.

**Comment by pozitronik** [comment]
Неужели они вернули WebDAV? Всего-то семь лет понадобилось.

**Comment by Immick** [comment]
А почему тут идет обсуждение плагина от WebDAV? Юзайте тогда уж их Диск-О - будет обычным диском в Тотале.
Лучше CloudMailRu для Mail.ru ничего нет. 

А WebDAV-плагин этот косячит: не смог прочитать путь с кириллицей - показался ему путь длинным.

**Comment by ViRT1ST** [comment]
> В общем, очень жду реанимации CloudMailRu со стороны Павла.

Я тоже жду. Работало все отлично годами, а тут облом от Mail.
Просто написал вариант, пока пересидеть временно.

**Comment by midi-creator** [comment]
Настроил webbdav? dcltkfk всё что мне нужно и решил опять проверить CloudMailRu - работает, что за? просто взяло и заработало?

**Comment by midi-creator** [comment]
или просто глюк на mail.ru был? а мы тут все перепугались...

**Comment by pozitronik** [comment]
> Настроил webbdav, сделал всё что мне нужно и решил опять проверить CloudMailRu - работает, что за? просто взяло и заработало?

Да, заработало. Саморассос 🤷‍♂️.

**Comment by Owyn** [comment]
С 2х ГБ до 1 ГБ лимит на файлы они уже скукожили походу :-(

**Comment by Immick** [comment]
@pozitronik, я вот понять не могу, за счет чего CloudMailRu заливает файлы в несколько раз быстрее, чем их фирменный Диск-О? Прямо в голове не укладывается. CloudMailRu же по http обычному аплоадит?

![image](https://github.com/user-attachments/assets/fffb4378-8028-436c-aeaa-6801a336f0a7)

Сверху - CloudMailRu, снизу - Диск-О

**Comment by kolokoltsov** [comment]
> Сверху - CloudMailRu, снизу - Диск-О. В чем фишка?... CloudMailRu же по http обычному аплоадит?
Потому что сверху у тебя не копирование, а подсчет хэша файла

**Comment by Immick** [comment]
> > Сверху - CloudMailRu, снизу - Диск-О. В чем фишка?... CloudMailRu же по http обычному аплоадит?
> 
> Потому что сверху у тебя не копирование, а подсчет хэша файла

Вместе с подсчетом кэша происходит копирование. Т.е. после окончания подсчета кэша файл уже фактически находится на облаке.
![2024-10-26_13-29-49](https://github.com/user-attachments/assets/3484e9c1-7002-4f94-904f-0bc04cc12548)
Как тебе такое, Илон?

**Comment by pasha00000** [comment]
@Immick если ранее любым пользователем в облако уже был загружен файл с таким хэшем как у вашего файла, то файл появится сразу после подсчёта. Если же файл уникальный и ранее не был в облаке, то после подсчёта хэша файл начнёт загружаться с обычной скоростью

**Comment by Owyn** [comment]
> > Настроил webbdav, сделал всё что мне нужно и решил опять проверить CloudMailRu - работает, что за? просто взяло и заработало?
> 
> Да, заработало. Саморассос 🤷‍♂️.

Раз теперь работает и "классика" и webdav - чем пользоваться через Total Commander? Есть различия (в функционале или ещё чём?)?

**Comment by Immick** [comment]
@Owyn 
1. Через WebDAV не читаются относительные длинные пути. По крайней мере на кириллице. С CloudMailRu такой проблемы нет.
2. Естественно, нет возможности расшаривать файлы.
3. Нет сплита файлов при загрузке.

**Comment by Snow2007** [comment]
У меня пишет такое:
EIdHTTPProtoclException error with message: HTTP/1.1 400 Bad Request at post data to https://cloud.mail.ru/home, 
server response....
error: parsing auth token for user@mail.ru

**Comment by eeak** [comment]
А у меня сегодня сломалось. Пересел на webdav, тоже ругалось на кирилицу, поставил галку utf-8 - работает.

**Comment by MihayloPupin** [comment]
У меня с самого начала установки плагина не заходил в облако с сообщением error: parsing auth token for.
При первом запуске вообще окно для ввода кода пропало. После перезагрузки окно появилось вновь.

Покопавшись в настройках безопасности аккаунта майл.ру нашёл резервные коды для двухфакторной аутентификации. При вводе этих кодов в окно всё подключается и работает.

Искать в веб-интерфейсе почты "Безопасность" - "Способы входа" - "Двухфакторная аутентификация" - "Резервные коды".
За раз на экран вываливают 10 кодов. Срок действия одного кода вроде как день. Может больше, просто чаще не заходил.

**Comment by mikleru** [comment]
Настроил два аккаунта: платный (рабочий) и бесплатный (личный). Платный работает, бесплатный - нет.

CONNECT \work@workdomain.ru
Login to work@workdomain.ru
Connected to work@workdomain.ru
Total space: 100 Gb, used: 24 Mb, free: 99 Gb.

CONNECT \user@mail.ru
Login to user@mail.ru
EIdHTTPProtocolException error with message: HTTP/1.1 400 Bad Request at post data to https://cloud.mail.ru/home, server response: <html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<hr><center>nginx</center>
</body>
</html>

error: parsing auth token for user@mail.ru

**Comment by pozitronik** [comment]
Закрыто: способ авторизации полностью изменён, см. https://github.com/pozitronik/CloudMailRu/issues/317
